runtime/src/bank.rs

```diff
@@ -3616,6 +3616,21 @@ impl Bank {
         )
         .unwrap();
         self.store_account_and_update_capitalization(&address, &new_account);
+        info!(
+            "create epoch_reward partition data account {} {address} {epoch_rewards_partition_data:#?}", self.slot
+        );
+
+        // ignore reward PDA account lamport when we are testing partitioned rewards before the actual feature activation.
+        if !self.is_partitioned_rewards_feature_enabled()
+            && (self
+                .partitioned_epoch_rewards_config()
+                .test_enable_partitioned_rewards
+                && self.get_reward_calculation_num_blocks() == 0
+                && self.partitioned_rewards_stake_account_stores_per_block() == u64::MAX)
+        {
+            self.capitalization
+                .fetch_sub(new_account.lamports(), Relaxed);
+        }
```

Comment:
That's an option, and probably the simplest option. However, this won't allow us to test the new reward rpc methods.

New reward rpc method require that those accounts exist in the account-db. The new rpc method, which will read these accounts and use them to extract stake rewards from the corresponding reward blocks.

Reply:
Good question. If the new RPC method is feature gated, then we won't be able to run the method without feature activation. 

But I think the RPC method will have no access to the feature gate. Instead, it relies on whether the PDA account exists or not to determine which code path to run. If the PDA account exists, then it will follow the code path that extracts rewards from partitioned reward blocks. Otherwise, it will follow the old code path to extract rewards from the first block in the epoch. 

@CriesofCarrots may confirm this?

Reply:
To clarify, it's not a new RPC method, just an adaptation of the existing `getInflationReward`. And yes, it checks whether the feature gate has been enabled to decide which block(s) to pull to find rewards. https://github.com/solana-labs/solana/pull/34773/files#diff-b55d9bcb5d2e7eae09fd352edcc765f9c2a4c05b97e9847582071d35850e8671R607
It will not know anything about the `--partitioned-epoch-rewards-force-enable-single-slot` flag. Since that flag puts all rewards in the first slot anyway, the RPC method should still work. But it's not a particularly useful test case for the RPC changes.

Reply:
OK. 
It looks like we are *unlikely* to use `--partitioned-epoch-rewards-force-enable-single-slot` as a way to test the `getInflationReward`  rpc.  This PR seems to be too "hacky". 

I will close this PR and create a new PR to get `--partitioned-epoch-rewards-force-enable-single-slot` working by simply not storing the new PDA account when the flag is active.